### ISSUE TITLE
Project updater removes assemblies and usages to namespaces other than xenko

### ISSUE THREAD

**Issue by tebjan** [xenko3d/xenko#33]
In a game project from xenko 2.1 i was referencing some other assemblies and used their namespaces in some files. after the upgrade the project did not have the references to the assemblies anymore and the usage section of the c# files was cleared from the other namespaces. in this case it was SharpDX assemblies.

**Comment by tebjan** [comment]
good point, since the updater removes all individual assemblies and replaces them with the single xenko package reference it seems natural that it removes the sharpdx. however, i just saw that there were other third party assemblies that have been removed too.

**Comment by xen2** [comment]
The upgrader recreate projects using new templates, and for simplicity it doesn't handle advanced cases.
So a few things are lost:
- Extra references
- New glob pattern includes everything, so most files should be reincluded, but you have to be careful because it might include some extra files or miss some that were outside the main folder
- AssemblyInfo is deleted (new project system should fill them)
- Special files with generators such as `.xksl` and Contents will be missing
- Any other csproj customization is lost.

However, since it is a one-time upgrade from 2.x to 3.x (from 3.x this shouldn't happen anymore), also it works fine for the standard cases and many people already did it anyway, I figured it was better to not spend too much time on that and focus on other things.

Sorry for the inconvenience.
Let me know if you really think that's a problem.
